tests/bluetooth/tester/src/btp_ccp.c

```diff
@@ -0,0 +1,367 @@
+				      const bt_tbs_client_call_state_t *call_states)
+{
+	struct net_buf_simple *buf = NET_BUF_SIMPLE(CALL_STATES_EV_SIZE);
+	struct btp_ccp_call_states_ev ev = {
+		sys_cpu_to_le32(err), inst_index, call_count
+	};
+
+	net_buf_simple_init(buf, 0);
+	net_buf_simple_add_mem(buf, &ev, sizeof(ev));
+
+	for (uint8_t n = 0; n < call_count; n++, call_states++) {
+		net_buf_simple_add_mem(buf, call_states, sizeof(bt_tbs_client_call_state_t));
+	}
+
+	tester_event(BTP_SERVICE_ID_CCP, BTP_CCP_EV_CALL_STATES, buf->data, buf->len);
+}
+
+/**
+ * @brief Callback function for ccp_read_call_state.
+ *
+ * @param conn          The connection used in the function.
+ * @param err           Error value. BT_TBS_CLIENT_RESULT_CODE_*,
+ *                      GATT error or errno value.
+ * @param inst_index    The index of the TBS instance that was updated.
+ * @param call_count    Number of call states read.
+ * @param call_states   Array of call states. The array is not kept by
+ *                      the client, so must be copied to be saved.
+ */
+static void tbs_client_call_states_cb(struct bt_conn *conn,
+				      int err,
+				      uint8_t inst_index,
+				      uint8_t call_count,
+				      const bt_tbs_client_call_state_t *call_states)
+{
+	LOG_DBG("Call states - err (%u) Call Count (%u)", err, call_count);
+
+	tbs_client_call_states_ev(err, inst_index, call_count, call_states);
+}
+
+/**
+ * @brief Callback function for ccp_read_termination_reason.
+ *
+ * @param conn          The connection used in the function.
+ * @param err           Error value. BT_TBS_CLIENT_RESULT_CODE_*,
+ *                      GATT error or errno value.
+ * @param inst_index    The index of the TBS instance that was updated.
+ * @param call_index    The call index.
+ * @param reason        The termination reason.
+ */
+static void tbs_client_termination_reason_cb(struct bt_conn *conn,
+					     int err,
+					     uint8_t inst_index,
+					     uint8_t call_index,
+					     uint8_t reason)
+{
+	LOG_DBG("Termination reason - err (%u) Call Index (%u) Reason (%u)",
+		err, call_index, reason);
+}
+
+static const struct bt_tbs_client_cb tbs_client_callbacks = {
+	.discover = tbs_client_discover_cb,
+	.originate_call = tbs_client_originate_call_cb,
+	.terminate_call = tbs_client_terminate_call_cb,
+	.accept_call = tbs_client_accept_call_cb,
+	.retrieve_call = tbs_client_retrieve_call_cb,
+	.call_state = tbs_client_call_states_cb,
+	.termination_reason = tbs_client_termination_reason_cb
+};
+
+static uint8_t ccp_register_callbacks(const void *cmd, uint16_t cmd_len,
+				      void *rsp, uint16_t *rsp_len)
+{
+	bt_tbs_client_register_cb(&tbs_client_callbacks);
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t ccp_discover_tbs(const void *cmd, uint16_t cmd_len,
+				void *rsp, uint16_t *rsp_len)
+{
+	const struct btp_ccp_discover_tbs_cmd *cp = cmd;
+	int err = (active_conn) ?
+		  bt_tbs_client_discover(active_conn, cp->subscribe) :
+		  -ENOTCONN;
```

Comment:
```suggestion
	int err = (active_conn) ? bt_tbs_client_discover(active_conn, cp->subscribe) : -ENOTCONN;
```

The maximum line length is 100, so this can fit on a single line

Reply:
OK.